pkg/migrations/migrations.go

```diff
@@ -17,7 +17,8 @@ type Operation interface {
 	// Start will apply the required changes to enable supporting the new schema
 	// version in the database (through a view)
 	// update the given views to expose the new schema version
-	Start(ctx context.Context, conn *sql.DB, stateSchema string, s *schema.Schema, cbs ...CallbackFn) error
+	// Returns the table that requires backfilling, if any.
+	Start(ctx context.Context, conn *sql.DB, stateSchema string, s *schema.Schema, cbs ...CallbackFn) (*schema.Table, error)
```

Comment:
wondering about the API here, not a big deal but returning a table can be strange unless you read this comment. We could consider creating a new struct to return here, mostly for DX reasons, thoughts?

Reply:
Is there a better alternative than a single-field struct? I'm not sure that provides a cleaner API or better DX.

Reply:
none than I can think of, I'm happy to iterate on this as we go